paradox-nakadi-consumer-core/src/main/java/de/zalando/paradox/nakadi/consumer/core/http/handlers/EventUtils.java

```diff
@@ -77,7 +81,7 @@ private EventUtils() { }
 
                 final ArrayNode arrayNode = (ArrayNode) eventsNode;
                 arrayNode.elements().forEachRemaining(element -> {
-                    if (!element.isNull()) {
+                    if (isEventOfValidType(element, eventType)) {
```

Comment:
[C] The delivery of an event with an unexpected type should cause an exception, or we risk losing events.